af/fastpath/fastpath/core.py

```diff
@@ -775,7 +775,8 @@ def score_measurement_whatsapp(msm):
             # has unexpected contents
             # Also note bug https://github.com/ooni/probe-legacy/issues/60
 
-        # TODO: handle elif url == "http://web.whatsapp.com/":
+        elif url == "http://web.whatsapp.com/":
+            webapp_accessible = b.get("failure", True) in (None, "", False)
```

Comment:
I think here you should set `webapp_accessible` equals to False if:

1. there is a failure, or
2. the status code is not 302, or
3. the number of location headers is != 1, or
4. the location is not the HTTPS URL